plugins/aggregators/histogram/histogram.go

```diff
@@ -151,6 +170,9 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 				index := sort.SearchFloat64s(buckets, value)
 				agr.histogramCollection[field][index]++
 			}
+			if !addTime.IsZero() {
+				agr.lastUpdateTime = addTime
```

Comment:
How about you store the `expireTime` here? This means we only need to calculate it once for every sample instead of computing it for each sample on each `Push`...

Reply:
nice improvement, will implement it